api/internal/localizer/localizer.go

```diff
@@ -127,3 +134,59 @@ func (lc *Localizer) localizeFile(path string) (string, error) {
 	}
 	return locPath, nil
 }
+
+// localizeBuiltinPlugins localizes built-in plugins on kust that can contain file paths. The built-in plugins
+// can be inline or in a file.
+func (lc *Localizer) localizeBuiltinPlugins(kust *types.Kustomization) error {
+	for fieldName, plugins := range map[string][]string{
+		"generator":   kust.Generators,
+		"transformer": kust.Transformers,
+		"validator":   kust.Validators,
```

Comment:
can we add unittests for `transformer` and `validator` as well?

Reply:
After attempting this, I decided to throw an error for `validators`. Because `alpha` only handles built-in transformers with file paths and all of such transformers must mutate their target resources (meaning they are invalid `validators`), `kustomize localize` will never localize any useful `validators` in `alpha`.

Reply:
I also decided to separate `generator` and `transformer` localization into 2 filters because it seemed strange that we'd allow `generators` like `ConfigMapGenerator` and `SecretGenerator` under `transformers` and vice versa.